src/main/java/com/microsoft/jenkins/azuread/AzureAdUser.java

```diff
@@ -84,11 +93,20 @@ public static AzureAdUser createFromJwt(JwtClaims claims) {
     }
 
     public void setAuthorities(Collection<ActiveDirectoryGroup> groups) {
-        GrantedAuthority[] newAuthorities = new GrantedAuthority[groups.size() * 2 + 2];
+        GrantedAuthority[] newAuthorities;
         int i = 0;
-        for (ActiveDirectoryGroup group : groups) {
-            newAuthorities[i++] = new AzureAdGroup(group.id(), group.name());
-            newAuthorities[i++] = new GrantedAuthorityImpl(group.id());
+        if (!groups.isEmpty()) {
```

Comment:
`groups` may be null.

Reply:
I didn't add a null check for `groups`, as the previous implementation also assumed `groups` to never be null.  See first line in old code: `GrantedAuthority[] newAuthorities = new GrantedAuthority[groups.size() * 2 + 2];`.
But if that is not the case, I can also add a null check at the beginning of the method.
Just let me know. And thanks for the quick review

Reply:
You are right. I should fix the potential risk by changing the code at:

https://github.com/jenkinsci/azure-ad-plugin/blob/f39cb0314f6c6789cd19647c5d159645e29f965d/src/main/java/com/microsoft/jenkins/azuread/AzureCachePool.java#L65-L68